src/ansys/dpf/core/vtk_helper.py

```diff
@@ -252,45 +251,88 @@ def compute_offset():
         """Return the starting point of a cell in the cells array"""
         return insert_ind + np.arange(insert_ind.size)
 
+    cells_insert_ind = compute_offset()
     # convert kAns to VTK cell type
     offset = None
     if as_linear:
+        # Map the vtk_cell_type to linear versions of the initial elements types
         vtk_cell_type = VTK_LINEAR_MAPPING[etypes]
 
-        # visualization bug within VTK with quadratic surf cells
-        ansquad8_mask = etypes == 6
-        if np.any(ansquad8_mask):  # kAnsQuad8
-
-            # simply copy the edge node indices to the midside points
-            offset = compute_offset()
-            cell_pos = offset[ansquad8_mask]
-            cells[cell_pos + 5] = cells[cell_pos + 1]
-            cells[cell_pos + 6] = cells[cell_pos + 2]
-            cells[cell_pos + 7] = cells[cell_pos + 3]
-            cells[cell_pos + 8] = cells[cell_pos + 4]
-
-        anstri6_mask = etypes == 4  # kAnsTri6 = 4
-        if np.any(anstri6_mask):
-            if offset is None:
-                offset = compute_offset()
-            cell_pos = offset[anstri6_mask]
-            cells[cell_pos + 4] = cells[cell_pos + 1]
-            cells[cell_pos + 5] = cells[cell_pos + 2]
-            cells[cell_pos + 6] = cells[cell_pos + 3]
+        # Create a global mask of connectivity values to take
+        mask = np.full(cells.shape, True)
+
+        # Get a mask of quad8 elements in etypes
+        quad8_mask = etypes == 6
+        # If any quad8
+        if np.any(quad8_mask):  # kAnsQuad8
+            # Get the starting indices of quad8 elements in cells
+            insert_ind_quad8 = cells_insert_ind[quad8_mask]
+            # insert_ind_quad8 += np.arange(insert_ind_quad8.size)
+            mask[insert_ind_quad8 + 5] = False
+            mask[insert_ind_quad8 + 6] = False
+            mask[insert_ind_quad8 + 7] = False
```

Comment:
@PProfizi as part of the development in TFS for the `mesh_to_pyvista`, I handled _all_ quadrratic eltypes, not only quads and shells. Thus, a `hex20` will be plotted as a `hex8`, a `wedge15` as a `wedge6` and so on and so forth. This indeed makes the cells structure very light for solid meshes. Do you consider handling it in Python as well?

Reply:
@rafacanton you are right, I should add those. I did limit myself to fixing what was already dealt with.

Reply:
@PProfizi We can cover it in a different PR, if you think that's better